hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -279,15 +279,37 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Set Ignition Server endpoint
 	{
+		serviceStrategy := servicePublishingStrategyByType(hcluster, hyperv1.Ignition)
+		if serviceStrategy == nil {
+			return ctrl.Result{}, fmt.Errorf("ignition server service strategy not specified")
+		}
 		controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
-		ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.GetName())
-		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionServerRoute), ignitionServerRoute); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return ctrl.Result{}, fmt.Errorf("failed to get ignitionServerRoute: %w", err)
+		switch serviceStrategy.Type {
+		case hyperv1.Route:
+			ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.GetName())
+			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionServerRoute), ignitionServerRoute); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return ctrl.Result{}, fmt.Errorf("failed to get ignitionServerRoute: %w", err)
+				}
 			}
-		}
-		if err == nil && ignitionServerRoute.Spec.Host != "" {
-			hcluster.Status.IgnitionEndpoint = ignitionServerRoute.Spec.Host
+			if err == nil && ignitionServerRoute.Spec.Host != "" {
+				hcluster.Status.IgnitionEndpoint = ignitionServerRoute.Spec.Host
+			}
+		case hyperv1.NodePort:
+			if serviceStrategy.NodePort == nil {
+				return ctrl.Result{}, fmt.Errorf("nodeport metadata not specified for ignition service")
+			}
+			ignitionService := ignitionserver.Service(controlPlaneNamespace.GetName())
+			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionService), ignitionService); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return ctrl.Result{}, fmt.Errorf("failed to get ignition service: %w", err)
+				}
+			}
+			if err == nil && len(ignitionService.Spec.Ports) == 1 && ignitionService.Spec.Ports[0].NodePort > 0 {
```

Comment:
nit: It'd be nice to move `len(ignitionService.Spec.Ports) == 1 && ignitionService.Spec.Ports[0].NodePort > 0` into a unit tested function e.g `ServiceNodePortAvailable` or so.

Reply:
fixing....

Reply:
fixed